cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -52,25 +52,64 @@ constexpr int32_t kBufferSizeLength = 4;
     throw ParquetException("Couldn't init ALG decryption");           \
   }
 
-class AesEncryptor::AesEncryptorImpl {
+class AesCryptoContext {
+ public:
+  AesCryptoContext(ParquetCipher::type alg_id, int32_t key_len, bool metadata,
+                   bool write_length) {
```

Comment:
This is used for reading and writing, so maybe `include_length` would be a better name?